openstack/resource_openstack_fw_policy_v2_test.go

```diff
@@ -0,0 +1,518 @@
+		}
+
+		var found *policies.Policy
+		for i := 0; i < 5; i++ {
+			// Firewall policy creation is asynchronous. Retry some times
+			// if we get a 404 error. Fail on any other error.
+			found, err = policies.Get(networkingClient, rs.Primary.ID).Extract()
+			if err != nil {
+				if _, ok := err.(gophercloud.ErrDefault404); ok {
+					time.Sleep(time.Second)
+					continue
+				}
+				return err
+			}
+			break
+		}
+
+		*policy = *found
+
+		return nil
+	}
+}
+
+const testAccFWPolicyV2Basic = `
+resource "openstack_fw_policy_v2" "policy_1" {}
+`
+
+const testAccFWPolicyV2Shared = `
+resource "openstack_fw_policy_v2" "policy_1" {
+  shared = true
+}
+`
+
+const testAccFWPolicyV2AddRules = `
+resource "openstack_fw_policy_v2" "policy_1" {
+  name        = "policy_1"
+  description = "terraform acceptance test"
+  audited     = true
+  rules       = [
+    "${openstack_fw_rule_v2.udp_deny.id}",
+    "${openstack_fw_rule_v2.tcp_allow.id}"
+  ]
+}
+
+resource "openstack_fw_rule_v2" "tcp_allow" {
+  protocol = "tcp"
+  action   = "allow"
+}
+
+resource "openstack_fw_rule_v2" "udp_deny" {
+  protocol = "udp"
+  action   = "deny"
+}
+`
+
+const testAccFWPolicyV2DeleteRules = `
+resource "openstack_fw_policy_v2" "policy_1" {
+  name        = "policy_1"
+  description = "terraform acceptance test"
+  rules       = [
+    "${openstack_fw_rule_v2.udp_deny.id}"
+  ]
+}
+
+resource "openstack_fw_rule_v2" "udp_deny" {
+  protocol = "udp"
+  action   = "deny"
+}
+`
+
+const testAccFWPolicyV2RulesOrderBasic = `
+resource "openstack_fw_rule_v2" "rule_1" {
+  protocol = "tcp"
+  action   = "deny"
+}
+
+resource "openstack_fw_rule_v2" "rule_2" {
+  protocol = "tcp"
+  action   = "allow"
+}
+
+resource "openstack_fw_rule_v2" "rule_3" {
+  protocol = "udp"
+  action   = "allow"
+}
+
+resource "openstack_fw_rule_v2" "rule_4" {
+  protocol = "udp"
+  action   = "deny"
+}
+
+resource "openstack_fw_policy_v2" "policy_1" {
+  name        = "policy_1"
+  description = "terraform acceptance test"
+  rules       = [
+    "${openstack_fw_rule_v2.rule_1.id}",
+	"${openstack_fw_rule_v2.rule_2.id}",
+	"${openstack_fw_rule_v2.rule_3.id}",
+	"${openstack_fw_rule_v2.rule_4.id}"
+  ]
+}
+
+data "openstack_fw_policy_v2" "policy_1" {
+  policy_id = "${openstack_fw_policy_v2.policy_1.id}"
+}
+`
+
+const testAccFWPolicyV2RulesOrderRemove = `
+resource "openstack_fw_rule_v2" "rule_1" {
+  protocol = "tcp"
+  action   = "deny"
+}
+
+resource "openstack_fw_rule_v2" "rule_2" {
+  protocol = "tcp"
+  action   = "allow"
+}
+
+resource "openstack_fw_rule_v2" "rule_3" {
+  protocol = "udp"
+  action   = "allow"
+}
+
+resource "openstack_fw_rule_v2" "rule_4" {
+  protocol = "udp"
+  action   = "deny"
+}
+
+resource "openstack_fw_policy_v2" "policy_1" {
+  name        = "policy_1"
+  description = "terraform acceptance test"
+  rules       = [
+    "${openstack_fw_rule_v2.rule_1.id}",
+	"${openstack_fw_rule_v2.rule_2.id}",
+	"${openstack_fw_rule_v2.rule_4.id}"
```

Comment:
I think you need to change the rule order here, not just removing the rule, e.g. 
```suggestion
	"${openstack_fw_rule_v2.rule_4.id}",
	"${openstack_fw_rule_v2.rule_2.id}",
        "${openstack_fw_rule_v2.rule_1.id}",
```

Reply:
nm, I found the correct order test

Reply:
sorry, but it's done))
```
=== RUN   TestAccFWPolicyV2_rulesOrder
--- PASS: TestAccFWPolicyV2_rulesOrder (17.09s)
```